### PR TITLE
Fixes CVC4 encoding of FPs

### PR DESCRIPTION
**Pull request by mikhailramalho**

### CODE REVIEW

File: src/solvers/cvc4/cvc_conv.cpp [hunk 1]
```diff
@@ -283,87 +285,116 @@ cvc_convt::mk_smt_fpbv_fma(smt_astt v1, smt_astt v2, smt_astt v3, smt_astt rm)
...
+      mrm->a,
+      mfrom->a),
+    mk_bv_sort(width));
 }
 
 smt_astt
 cvc_convt::mk_smt_typecast_from_fpbv_to_sbv(smt_astt from, std::size_t width)
 {
-  smt_sortt to = mk_bv_sort(width);
+  const cvc_smt_ast *mrm =
```
Comment (lucasccordeiro) [inline src/solvers/cvc4/cvc_conv.cpp:308]: Some comment here would be helpful.

File: src/solvers/smt/fp/fp_conv.cpp [hunk 2]
```diff
@@ -23,6 +23,25 @@ static smt_astt extract_exp_sig(smt_convt *ctx, smt_astt fp)
...
+static BigInt power2(std::size_t n, bool negated)
+{
+  BigInt b;
+  b.setPower2(n);
+  if(negated)
+    b.negate();
+  return b;
+}
+
+static BigInt power2m1(std::size_t n, bool negated)
```
Comment (lucasccordeiro) [inline src/solvers/smt/fp/fp_conv.cpp:35]: Some comment here would be helpful.

**Review by lucasccordeiro — APPROVED** [review]
Some comments in the code of this PR would be helpful.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -29,15 +29,8 @@ jobs:
...
-    - name: Get older cvc4 build
-      id: cache-cvc4
-      uses: actions/cache@v1
-      with:
-        path: cvc4
-        key: ubuntu-cvc4
     - name: Setup CVC4
-      if: steps.cache-cvc4.outputs.cache-hit != 'true'          
-      run: wget https://github.com/CVC4/CVC4/archive/1.7.tar.gz && tar xf 1.7.tar.gz && rm 1.7.tar.gz && cd CVC4-1.7 && ./contrib/get-antlr-3.4 && ./configure.sh --optimized --prefix=../cvc4 --static --no-static-binary && cd build && make -j8 && make install
+      run: git clone https://github.com/CVC4/CVC4.git && cd CVC4 && ./contrib/get-antlr-3.4 && ./configure.sh --optimized --prefix=../cvc4 --static --no-static-binary && cd build && make -j8 && make install
```
Comment (rafaelsamenezes) [inline .github/workflows/build.yml]: Is there a reason to not use a fixed version or commit? This could affect the stability of the CI

**Review by rafaelsamenezes — APPROVED** [review]
LGTM

**Pull request merged**